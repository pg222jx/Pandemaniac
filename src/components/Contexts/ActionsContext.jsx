import React,{ useState, createContext } from 'react';

export const ActionsContext = createContext()

export const ActionsProvider = props => {

    const [amount, setAmount] = useState(2)

    return (
        <ActionsContext.Provider value={[amount, setAmount]}>
            {props.children}
        </ActionsContext.Provider>
    )
}

