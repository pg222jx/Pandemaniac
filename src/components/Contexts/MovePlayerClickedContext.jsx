import React,{ useState, createContext } from 'react';

export const MovePlayerClickedContext = createContext()

export const MovePlayerClickedProvider = props => {

    const [clicked, setClicked] = useState(false)

    return (
        <MovePlayerClickedContext.Provider value={[clicked, setClicked]}>
            {props.children}
        </MovePlayerClickedContext.Provider>
    );
}

