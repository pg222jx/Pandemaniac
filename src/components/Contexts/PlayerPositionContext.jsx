import React,{ useState, createContext } from 'react';

export const PlayerPositionContext = createContext()

export const PlayerPositionProvider = props => {

    const [position, setPosition] = useState({bottom: undefined, left: undefined})

    return (
        <PlayerPositionContext.Provider value={[position, setPosition]}>
            {props.children}
        </PlayerPositionContext.Provider>
    )
}

