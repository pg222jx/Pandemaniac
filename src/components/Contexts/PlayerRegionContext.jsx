import React,{ useState, createContext } from 'react';

export const PlayerRegionContext = createContext()

export const PlayerRegionProvider = props => {

    const [region, setRegion] = useState('')

    return (
        <PlayerRegionContext.Provider value={[region, setRegion]}>
            {props.children}
        </PlayerRegionContext.Provider>
    )
}

