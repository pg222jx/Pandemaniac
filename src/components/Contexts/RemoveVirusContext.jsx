import React,{ useState, createContext } from 'react';

export const RemoveVirusContext = createContext()

export const RemoveVirusProvider = props => {

    const [clicked, setClicked] = useState(false)

    return (
        <RemoveVirusContext.Provider value={[clicked, setClicked]}>
            {props.children}
        </RemoveVirusContext.Provider>
    );
}

