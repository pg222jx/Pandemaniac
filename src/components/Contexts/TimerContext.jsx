import React,{ useState, createContext } from 'react';

export const TimerContext = createContext()

export const TimerProvider = props => {

    let [time, setTime] = useState(0)

    return (
        <TimerContext.Provider value={[time, setTime]}>
            {props.children}
        </TimerContext.Provider>
    )
}

