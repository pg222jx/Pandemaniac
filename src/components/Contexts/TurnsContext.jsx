import React,{ useState, createContext } from 'react';

export const TurnsContext = createContext()

export const TurnsProvider = props => {

    const [amount, setAmount] = useState(0)

    return (
        <TurnsContext.Provider value={[amount, setAmount]}>
            {props.children}
        </TurnsContext.Provider>
    )
}

