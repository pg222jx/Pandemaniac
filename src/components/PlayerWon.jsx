/**
 * Representing the 'Start Game' button.
 *
 * @author Pernilla GÃ¶th
 * @version 1.0.0
 */


import React, { useEffect, useContext } from 'react'
import HighscoreForm from './HighscoreForm'
import { TurnsContext } from './Contexts/TurnsContext'
import { TimerContext } from './Contexts/TimerContext'


const GameOver = (props) => {
    const [turns, setTurns] = useContext(TurnsContext)// eslint-disable-line no-unused-vars
    const [time, setTime] = useContext(TimerContext)// eslint-disable-line no-unused-vars

    /**
    * Converts number to minutes and seconds.
    * 
    * @returns {string} - Representing the time in minutes and seconds.
    */
    const convertTime = () => {
        const totalSeconds = time
        const minutes = Math.floor(totalSeconds/60)
        const seconds = totalSeconds - minutes * 60
        return minutes + ':' + seconds
    }

    /**
    * Refreshes the page on Try Again.
    */
    const refreshPage = () => {
        window.location.reload(false)
    }

    /**
    * UseEffect hook to stop timer and hide Game components on render. 
    */
    useEffect(() => {
        props.onRender()
        props.stopTimer()
    })

    return (
        <React.Fragment>
        <div className="playerWon">
            <h3>Congratulations!</h3>
            <p>You have now built a science centre in every continent and</p>
            <p>therefore saved the world in {turns} turns and with the time: {convertTime()}!</p>
            <HighscoreForm/>
            <button className="tryAgain" onClick={refreshPage}>Try again?</button>
            </div>
        </React.Fragment>
    )
}
 
export default GameOver