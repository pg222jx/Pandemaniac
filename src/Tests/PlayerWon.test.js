import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import { TurnsProvider } from '../components/Contexts/TurnsContext'
import { TimerProvider } from '../components/Contexts/TimerContext'
import PlayerWon from '../components/PlayerWon'


const wrapperShallow = shallow(
    <TurnsProvider>
        <TimerProvider>
            <PlayerWon/>
        </TimerProvider>
    </TurnsProvider>
)


describe('testing the PlayerWon component', () => {
    it('should match the snapshot', () => {
        expect(wrapperShallow.html()).toMatchSnapshot()
      })

      it('should render without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(
            <TurnsProvider>
                <TimerProvider>
                    <PlayerWon/>
                </TimerProvider>
            </TurnsProvider>, div
        )
       })
})
