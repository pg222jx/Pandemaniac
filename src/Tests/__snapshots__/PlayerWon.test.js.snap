// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the PlayerWon component should match the snapshot 1`] = `"<div class=\\"playerWon\\"><h3>Congratulations!</h3><p>You have now built a science centre in every continent and</p><p>therefore saved the world in 0 turns and with the time: 0:0!</p><div class=\\"highscore_form_container\\"><form class=\\"highscore_form\\"><label>Please enter a nickname for the highscore list:</label><br/><br/><input type=\\"text\\" class=\\"enterNickname\\" value=\\"\\" placeholder=\\"Enter nickname\\"/><p class=\\"name_info\\" data-testid=\\"name_info_test\\"></p><input type=\\"submit\\" data-testid=\\"submit_test\\" class=\\"submit\\" value=\\"Submit\\"/></form><p class=\\"score_info\\" data-testid=\\"score_info_test\\"></p></div><button class=\\"tryAgain\\">Try again?</button></div>"`;
